pkg/kv/kvserver/replica_lifecycle_datadriven_test.go

```diff
@@ -426,6 +481,16 @@ func (r *replicaInfo) initialized() bool {
 	return r.hs.Commit > 0 // NB: or r.ts.Index > 0
 }
 
+// pendingSplit represents a split that has been evaluated but not yet applied.
+type pendingSplit struct {
+	trigger *roachpb.SplitTrigger
+	batch   storage.Batch
```

Comment:
The `storage.Batch` may hold iterators and stuff, pinning Pebble. Should we make it close to the real raft proposal, in which there is an encoded batch? Something like:

```golang
type pendingSplit struct {
	trigger roachpb.SplitTrigger
	batch []byte // encoded WriteBatch.Repr()
}
```

We can then, in the apply side, write this repr to a `Writer` with `Writer.ApplyBatchRepr()`.